### PR TITLE
Feature/consumers

### PR DESCRIPTION
**Pull request by Taliik**
## Summary of the changes / Why this is an improvement
- removed product deploy and consumer-sets list | edit
- added consumers deploy | edit | list and the corresponding documentation

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: croud/consumers/commands.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    query.execute(vars)
+    print_query(query, "deployConsumer")
+
+
+def consumers_list(args: Namespace) -> None:
+    body = dedent(
+        """
+    query allConsumers($clusterId: ID, $productName: String, $projectId: ID) {
+        allConsumers(clusterId: $clusterId, productName: $productName, projectId: $projectId) {
+            id
```
Comment (chaudum) [inline croud/consumers/commands.py]: We have the following fields available:
```
    id = graphene.ID()
    name = graphene.String()
    project_id = graphene.ID()
    project = graphene.Field(Project)
    cluster_id = graphene.ID()
    cluster = graphene.Field(Cluster)
    product_name = graphene.String()
    product_tier = graphene.String()
    instances = graphene.Int()
    table_name = graphene.String()
    table_schema = graphene.String()
```

File: croud/consumers/commands.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+from croud.gql import Query, print_query
+from croud.util import clean_dict
+
+
+def consumers_deploy(args: Namespace) -> None:
+    body = dedent(
+        """
+    query deployConsumer($input: DeployConsumerInput) {
+        deployConsumer(input: $input) {
+            id
```
Comment (chaudum) [inline croud/consumers/commands.py]: What fields should we query. Fields see in comment below.

Comment (MarkusH) [reply croud/consumers/commands.py]: I'd go with id, cluster_id, project_id, schema, table, product, tier, instances

Comment (chaudum) [reply croud/consumers/commands.py]: Sounds good to me

File: croud/consumers/commands.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    query = Query(body, args)
+    query.execute(vars)
+    print_query(query, "allConsumers")
+
+
+def consumers_edit(args: Namespace) -> None:
+    body = dedent(
+        """
+    mutation editConsumer($id: ID!, $input: EditConsumerInput!) {
+        editConsumerSet(id: $id, input: $input) {
```
Comment (m-kharbat) [inline croud/consumers/commands.py]: editConsumerSet > editConsumer

File: croud/consumers/commands.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+from textwrap import dedent
+
+from croud.gql import Query, print_query
+from croud.util import clean_dict
+
+
+def consumers_deploy(args: Namespace) -> None:
+    body = dedent(
+        """
+    query deployConsumer($input: DeployConsumerInput) {
```
Comment (m-kharbat) [inline croud/consumers/commands.py]: `$input: DeployConsumerInput` is a required field > ` $input: DeployConsumerInput!`

File: tests/unit_tests/test_commands.py [hunk 5]
```diff
@@ -567,200 +567,173 @@ def test_list_filter_conflict(self, mock_run, mock_load_config):
...
+    def test_consumers_edit(self, mock_run, mock_load_config):
         expected_body = textwrap.dedent(
             """
-    mutation editConsumerSet($id: String!, $input: EditConsumerSetInput!) {
-        editConsumerSet(
-            id: $id,
-            input: $input
-        ) {
+    mutation editConsumer($id: ID!, $input: EditConsumerInput!) {
+        editConsumerSet(id: $id, input: $input) {
```
Comment (m-kharbat) [inline tests/unit_tests/test_commands.py]: editConsumerSet > editConsumer

File: croud/consumers/commands.py (hunk 4 above)
Comment (m-kharbat) [inline croud/consumers/commands.py]: this should be a mutation

File: croud/consumers/commands.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+    query = Query(body, args)
+    query.execute(vars)
+    print_query(query, "allConsumers")
+
+
+def consumers_edit(args: Namespace) -> None:
+    body = dedent(
+        """
+    mutation editConsumer($id: ID!, $input: EditConsumerInput!) {
```
Comment (m-kharbat) [inline croud/consumers/commands.py]: ID! >  String!

Comment (chaudum) [reply croud/consumers/commands.py]: Should we change that in the backend?

**Pull request merged**